fe/fe-core/src/main/java/org/apache/doris/analysis/SelectStmt.java

```diff
@@ -631,6 +631,24 @@ public void analyze(Analyzer analyzer) throws UserException {
             tableRef.setOnClause(rewriteQueryExprByMvColumnExpr(tableRef.getOnClause(), analyzer));
         }
 
+        // if the query contains limit clause but not order by clause, order by keys by default.
+        if (ConnectContext.get() != null && ConnectContext.get().getSessionVariable() != null
+                && ConnectContext.get().getSessionVariable().isEnableDefaultOrder() && analyzer.isNeedAddOrderBy()) {
+            orderByElements = Lists.newArrayList();
+            for (TableRef ref : getTableRefs()) {
+                if (ref instanceof BaseTableRef
+                        && ref.getTable().getType() == TableType.OLAP) {
+                    OlapTable olapTable = ((OlapTable) ref.getTable());
+                    int num = olapTable.getKeysNum();
+                    for (int i = 0; i < num; ++i) {
+                        String colName = olapTable.getFullSchema().get(i).getName();
+                        SlotRef slotRef = new SlotRef(null, colName);
+                        orderByElements.add(new OrderByElement(slotRef, true, true));
```

Comment:
i think it will failed when we have agg, window function, ...
we only process SPJ is enough.